source/adapters/hip/command_buffer.cpp

```diff
@@ -9,156 +9,756 @@
 //===----------------------------------------------------------------------===//
 
 #include "command_buffer.hpp"
+
+#include <hip/hip_runtime.h>
+
 #include "common.hpp"
+#include "enqueue.hpp"
+#include "event.hpp"
+#include "kernel.hpp"
+#include "memory.hpp"
+#include "queue.hpp"
+
+#include <cstring>
+
+ur_exp_command_buffer_handle_t_::ur_exp_command_buffer_handle_t_(
+    ur_context_handle_t hContext, ur_device_handle_t hDevice)
+    : Context(hContext),
+      Device(hDevice), HIPGraph{nullptr}, HIPGraphExec{nullptr}, RefCount{1} {
+  urContextRetain(hContext);
+  urDeviceRetain(hDevice);
+}
+
+/// The ur_exp_command_buffer_handle_t_ destructor releases
+/// all the memory objects allocated for command_buffer managment
+ur_exp_command_buffer_handle_t_::~ur_exp_command_buffer_handle_t_() {
+  // Release the memory allocated to the Context stored in the command_buffer
+  UR_TRACE(urContextRelease(Context));
+
+  // Release the device
+  UR_TRACE(urDeviceRelease(Device));
+
+  // Release the memory allocated to the HIPGraph
+  UR_CHECK_ERROR(hipGraphDestroy(HIPGraph));
+
+  // Release the memory allocated to the HIPGraphExec
+  UR_CHECK_ERROR(hipGraphExecDestroy(HIPGraphExec));
+}
+
+/// Helper function for finding the HIP Nodes associated with the
```

Comment:
Seems like this could be re-formatted:
```
/// Helper function for finding the HIP Nodes associated with the commands in a
/// command-buffer, each event is pointed to by a sync-point in
```

Reply:
This has be re-formatted.